src/Nethermind/Nethermind.BeaconNode.Peering/RpcMessage_T.cs

```diff
@@ -0,0 +1,34 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using Nethermind.Core2.P2p;
+
+namespace Nethermind.BeaconNode.Peering
+{
+    public class RpcMessage<T>
+    {
+        public RpcMessage(string peerId, RpcDirection direction, T content)
+        {
+            PeerId = peerId;
+            Direction = direction;
+            Content = content;
+        }
+
+        public T Content { get; }
+        public RpcDirection Direction { get; }
```

Comment:
interesting, is it needed on the message itself?

Reply:
Maybe it is bad naming... Content is the actual message (sent on the wire), but there are two different functions in Mothra for send request and send response; and when it is received there is a flag for request/response (but doesn't seem to be working). It is more a Mothra thing, I think.

For something like get blocks the calls are non-symmetrical (request one way, blocks the other), so the different directions are obvious. For Status, however, they are the same in both directions.

I'll review the Eth1 code, but if you have a suggested alternative name, let me know.  If it is only status that this affects, I might just special case it.

Reply:
I'm going to leave this for now, and once I've done some other RPC calls work out if status is a special case and can be done differently.